service/frontend/api/handler.go

```diff
@@ -1174,6 +1178,7 @@ func (wh *WorkflowHandler) RespondActivityTaskFailed(
 		sizeLimitWarn,
 		sizeLimitError,
 		taskToken.DomainID,
+		domainName,
```

Comment:
really makes me wish we had a common name-and-id pair :|  we have access to both almost everywhere...

Reply:
Yep. We do have a domain cache, but it is not propagated everywhere. I f we used DI, we could separate this component and make sure that domain cache is provided, but we don't for now.